Fix/address ledger audit vuln

keiff3r opened the pull request.

reffiek committed:
fix: resolve possible out of bound issue

reffiek committed:
fix: patch potential integer underflow

reffiek committed:
fix: disable debuging

reffiek committed:
fix: add a check before using variable

reffiek committed:
fix: handle weakness

reffiek committed:
fix: add a condition in display_data_content_and_confirm

reffiek committed:
chore: bump version

reffiek committed:
fix: remove compilation warning

reffiek committed:
fix: always include debug helpers header

- Move debug-helpers/debug.h inclusion out of DEBUG=1 condition
- Keep -g flag only for debug builds

reffiek committed:
test: generate golden snapshots

n4l5u0r commented on Makefile:
```diff
@@ -159,7 +159,7 @@ DEFINES   += HAVE_BOLOS_APP_STACK_CANARY
 
 # debugging helper functions and macros
-CFLAGS    += -g -include debug-helpers/debug.h
+CFLAGS    += -include debug-helpers/debug.h
```
@keiff3r we need to condition the whole part to be included in the 
``` 
ifeq ($(DEBUG),1)
    CFLAGS    += -g
endif```

n4l5u0r commented on tests/snapshots/flex/test_dashboard/00001.png:
@keiff3r copyright to change to `Acre` instead of `Blooo`

n4l5u0r reviewed (changes requested):
Please implement the changes

reffiek committed:
style: change copyright from Blooo to Acre

- change 'Blooo' to 'Acre'
- generate golden snapshots

reffiek committed:
chore: update Makefile for debug configuration

- Remove inclusion of debug-helpers/debug.h from default CFLAGS.
- Add inclusion of debug-helpers/debug.h to CFLAGS only when DEBUG=1.
- Ensure CFLAGS includes -g for debugging when DEBUG=1.

n4l5u0r committed:
fix: added missing refs for debug tools

reffiek committed:
style: change developer from Blooo to Acre

- change "Blooo" to "Acre"
- generate golden snapshots
